### PR TITLE
Various spec fixes for EIP-7251

### PR DESCRIPTION
**Pull request by ralexstokes**
Spec fixes found while working on #3656.

Will merge here, and rebase that to just have test fixes.

### CODE REVIEW

**Review by hwwhww — APPROVED** [review]
I guess the bugs were primarily caused by that the previous pre-fork was Bellatrix.

This fix looks good to me! (though, we need #3656 to verify it 😄)

File: specs/_features/eip7251/beacon-chain.md [hunk 1]
```diff
@@ -989,12 +1055,70 @@ def process_voluntary_exit(state: BeaconState, signed_voluntary_exit: SignedVolu
...
+    # Fill in sync committees
+    # Note: A duplicate committee is assigned for the current and next committee at genesis
+    state.current_sync_committee = get_next_sync_committee(state)
+    state.next_sync_committee = get_next_sync_committee(state)
+
+    # Initialize the execution payload header
+    # If empty, will initialize a chain that has not yet gone through the Merge transition
+    state.latest_execution_payload_header = execution_payload_header
+
+    return state
```
Comment (dapplion) [inline specs/_features/eip7251/beacon-chain.md:1123]: @hwwhww are we still merging these? state bumps can be computed from the upgrade_to_* function

Comment (ralexstokes) [reply specs/_features/eip7251/beacon-chain.md:1123]: they do serve distinct purposes

for testing, we will usually start at genesis, whereas at the fork boundary there is likely state we want to bring forward/translate from the old fork to the new

Comment (hwwhww) [reply specs/_features/eip7251/beacon-chain.md:1123]: I think @dapplion was pointing to an old discussion.  I just opened an issue: #3663 (a good first issue!)

It's better to clean up them in other PRs altogether. 👍

**Pull request merged**